app/src/util/navigation.ts

```diff
@@ -1,12 +1,20 @@
 import { createNavigationContainerRef } from '@react-navigation/native';
 import { type StackParamList } from '../screens/types';
+import { setInterval } from './setTimeout';
 
 export const navigationRef = createNavigationContainerRef<StackParamList>();
 
-export function getNavigate() {
-  if (!navigationRef.isReady()) {
-    return;
-  }
+export async function getNavigate() {
+  await new Promise<void>((resolve) => {
+    const poll = () => {
+      if (navigationRef.isReady()) {
+        clearInterval(intervalId);
+        resolve();
+      }
+    };
+    const intervalId = setInterval(poll, 500);
```

Comment:
What made you choose 500ms as the interval?

Reply:
No particular reason. It seemed short enough that the UX wouldn't be too negatively impacted and long enough as to not overpoll and cause performance issues.